manifests/init.pp

```diff
@@ -10,3 +10,14 @@
 # it and/or modify it under the terms of the GNU
 # General Public License version 3 as published by
 # the Free Software Foundation.
+
+class user (
+  $managed_hash = {},
+  ) {
+
+  if $managed_hash != {} {
```

Comment:
To keep some coherency with other similar cases, I'd call this parameter $users_hash or $managed_users_hash.
The rest is ok for me